offgridplanner/optimization/helpers.py

```diff
@@ -271,3 +271,321 @@ def check_imported_demand_data(df, project_dict):
+                },
+                "consumer_detail": {
+                    "type": "object",
+                    "additionalProperties": {"type": "string"},
+                },
+                "is_connected": {
+                    "type": "object",
+                    "additionalProperties": {"type": "boolean"},
+                },
+            },
+            "additionalProperties": False,
+        },
+        "grid_design": {
+            "type": "object",
+            "required": ["distribution_cable", "connection_cable", "pole", "mg", "shs"],
+            "properties": {
+                "distribution_cable": {
+                    "type": "object",
+                    "required": ["capex", "max_length", "epc"],
+                    "properties": {
+                        "capex": {"type": "number"},
+                        "max_length": {"type": "number"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "connection_cable": {
+                    "type": "object",
+                    "required": ["capex", "max_length", "epc"],
+                    "properties": {
+                        "capex": {"type": "number"},
+                        "max_length": {"type": "number"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "pole": {
+                    "type": "object",
+                    "required": ["capex", "max_n_connections", "epc"],
+                    "properties": {
+                        "capex": {"type": "number"},
+                        "max_n_connections": {"type": "integer"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "mg": {
+                    "type": "object",
+                    "required": ["connection_cost", "epc"],
+                    "properties": {
+                        "connection_cost": {"type": "number"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "shs": {
+                    "type": "object",
+                    "required": ["include", "max_grid_cost"],
+                    "properties": {
+                        "include": {"type": "boolean"},
+                        "max_grid_cost": {"type": "number"},
+                    },
+                },
+            },
+        },
+        "yearly_demand": {"type": "number"},
+    },
+    "additionalProperties": False,
+}
+
+GRID_V2_SCHEMA = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "required": ["node_fields", "nodes", "grid_design", "yearly_demand"],
+    "properties": {
+        "node_fields": {"type": "array", "items": {"type": "string"}},
+        "nodes": {
+            "type": "array",
+            "items": {
+                "type": "array",
+                "items": [
```

Comment:
The "enum" fields below need to be reviewed and expanded as they were created based on a dummy example